### PR TITLE
Detach fval in torch_minimize to avoid memory leak

### PR DESCRIPTION
**Pull request by mikkelbue**
## Motivation

This fixes https://github.com/pytorch/botorch/issues/2526

### Have you read the [Contributing Guidelines on pull requests](https://github.com/pytorch/botorch/blob/main/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan
The tests are passing.

## Related PRs
N/A

### CODE REVIEW

File: botorch/optim/core.py [hunk 1]
```diff
@@ -202,7 +202,7 @@ def torch_minimize(
         )
 
         # TODO: Update stopping_criterion API to return a message.
-        if stopping_criterion and stopping_criterion(fval):
+        if stopping_criterion and stopping_criterion(fval.detach()):
```
Comment (esantorella) [inline botorch/optim/core.py:205]: It looks like `fval` is detached everywhere it's used. Perhaps line 195 should be `fval = closure()[0].detach()`?

Comment (Balandat) [reply botorch/optim/core.py:205]: yeah that would be better

Comment (mikkelbue) [reply botorch/optim/core.py:205]: Good idea.

**Review by esantorella — APPROVED** [review]
Thanks!
